sympy/physics/mechanics/jointsmethod.py

```diff
@@ -0,0 +1,222 @@
+from sympy.physics.mechanics import Body, Lagrangian, KanesMethod, LagrangesMethod
+from sympy.physics.mechanics.method import _Methods
+
+__all__ = ['JointsMethod']
+
+
+class JointsMethod(_Methods):
+    """Method for formulating the equations of motion using a set of interconnected bodies with joints.
+
+    Parameters
+    ==========
+
+    newtonion : Body or ReferenceFrame
+        The newtonion(inertial) frame.
+    *joints : Joint
+        The joints in the system
+
+    Attributes
+    ==========
+
+    q, u : Matrix, shape(n, 1)
+        Matrices of the generalized coordinates and speeds
+    bodies : iterable
+        Iterable of Body objects in the system.
+    forcelist : iterable
+        Iterable of (Point, vector) or (ReferenceFrame, vector) tuples
+        describing the forces on the system.
+    mass_matrix : Matrix, shape(n, n)
+        The system's mass matrix
+    forcing : Matrix
+        The system's forcing vector
+    mass_matrix_full : Matrix
+        The "mass matrix" for the u's and q's
+    forcing_full : Matrix
+        The "forcing vector" for the u's and q's
+    method : KanesMethod or Lagrange's method
+        Method's object.
+
+    Examples
+    ========
+
+    This is a simple example for a one degree of freedom translational
+    spring-mass-damper.
+
+    >>> from sympy import symbols
+    >>> from sympy.physics.mechanics import Body, JointsMethod, PrismaticJoint
+    >>> from sympy.physics.vector import dynamicsymbols
+    >>> c, k = symbols('c k')
+    >>> q, u = dynamicsymbols('q u')
+    >>> W = Body('W')
+    >>> B = Body('B')
+    >>> J = PrismaticJoint('J', W, B, coordinates=q, speeds=u)
+    >>> W.apply_force(c*u*W.x, reaction_body=B)
+    >>> W.apply_force(k*q*W.x, reaction_body=B)
+    >>> method = JointsMethod(W, J)
+    >>> method.form_eoms()
+    (Matrix([[-c*u(t) - k*q(t)]]), Matrix([[-B_mass*Derivative(u(t), t)]]))
+    >>> MM = method.mass_matrix_full
+    >>> forcing = method.forcing_full
+    >>> rhs = MM.LUsolve(forcing)
+    >>> rhs
+    Matrix([
+    [                     u(t)],
+    [(-c*u(t) - k*q(t))/B_mass]])
+
+    Notes
+    =====
+
+    ``JointsMethod`` currently only works with systems that do not have any
+    configuration or motion constraints.
+
+    """
+
+    def __init__(self, newtonion, *joints):
+        if isinstance(newtonion, Body):
+            self.frame = newtonion.frame
+        else:
+            self.frame = newtonion
+
+        self._joints = joints
+        self._bodylist = self._generate_bodylist()
+        self._loadlist = self._generate_loadlist()
+        self._q = self._generate_q()
+        self._u = self._generate_u()
+        self._kdes = self._generate_kdes()
+
+        self._method = None
+
+    @property
+    def bodies(self):
+        return self._bodylist
+
+    @property
+    def forcelist(self):
+        return self._loadlist
+
+    @property
+    def q(self):
```

Comment:
All these properties should be documented with docstrings (rather than manually in the class docstring). They sphinx's autdoc will pick them up.

```python
@property
def q(self):
    """List of the generalized coordinates."""
```

Reply:
Added